src/stringobj.c

```diff
@@ -1267,7 +1268,7 @@ Obj CopyToStringRep(
 */
 Obj ImmutableString(Obj string)
 {
-    if (!IS_STRING_REP(string) || !IS_MUTABLE_OBJ(string)) {
+    if (!IS_STRING_REP(string) || IS_MUTABLE_OBJ(string)) {
```

Comment:
This one is much more recently, from 742f1a4 this September (and also my fault... :-/).  On the upside, it is not in GAP 4.10, so not much harm done :-).

Reply:
It is also not a critical issue. `ImmutableString()` is only used for the name field of functions and methods, meaning that in the worst case, that name was still mutable. Again, I ran into it during testing HPCGAP, where it led to names being in a thread-local region and thus inaccessible from other threads.